tool/tsh/common/tsh.go

```diff
@@ -3349,6 +3341,13 @@ func retryWithAccessRequest(
 		return trace.Wrap(origErr)
 	}
 
+	// Print and log the original AccessDenied error.
+	fmt.Fprintln(os.Stderr, utils.UserMessageFromError(origErr))
+	fmt.Fprintf(os.Stdout, "You do not currently have access to %q, attempting to request access.\n\n", resource)
+	if err := promptUserForAccessRequestDetails(cf, req); err != nil {
+		return trace.Wrap(err)
+	}
+
 	if err := setAccessRequestReason(cf, req); err != nil {
```

Comment:
I would handle `setAccessRequestReason` from within `promptUserForAccessRequestDetails`, logically it's part of that